[Feature Request] Simple way to extend/override TypeOrmCrudService

Hi,

I am trying to create a custom TypeOrmCrudService for my application, overriding the behaviour of the different methods...

To do this, I am basing my override methods on the current implementations on TypeOrmCrudService. Thing is, there is a lot of really useful methods on the base class (ex getOneOrFail(), prepareEntityBeforeSave(), etc)  , but I cannot use them on my override method, because they have private visibility... 

Would it be possible to make those methods protected instead, to be able to call them from derived classes? Or do you feel this would make the interface harder to maintain? If not, what would be a good way to be able to use those methods?

Thanks,

Hi @jbjhjm. I already tried this. But the event is not fired.
I suppose I reach this issue and maybe it has not been published yet in stable version : 
https://github.com/nestjsx/crud/issues/51

I tried to upgrade to 4.3.0-alpha.0 (upgrading crud, crud-typeorm and crud-request but I have this typescript error : 
```
node_modules/@nestjsx/crud-request/lib/types/request-query.types.d.ts:62:5 - error TS2411: Property '$and' of type 'undefined' is not assignable to string index type 'string | number | boolean | SFieldOperator | (SFields | SConditionAND)[]'.

62     $and?: never;
```

But I must says that I'm not fan of this method anyway. I think that every interaction with entity must be centralized in service, dao or repository. This is why being able to override prepareEntityBeforeSave would be a greater option imho.

--- a/packages/crud-typeorm/src/typeorm-crud.service.ts
+++ b/packages/crud-typeorm/src/typeorm-crud.service.ts
@@ -432,7 +432,7 @@ export class TypeOrmCrudService<T> extends CrudService<T> {
     );
   }
 
-  private async getOneOrFail(req: CrudRequest): Promise<T> {
+  protected async getOneOrFail(req: CrudRequest): Promise<T> {
     const { parsed, options } = req;
     const builder = await this.createBuilder(parsed, options);
     const found = await builder.getOne();
@@ -444,7 +444,7 @@ export class TypeOrmCrudService<T> extends CrudService<T> {
     return found;
   }
 
-  private async getOneShallowOrFail(where: ObjectLiteral): Promise<T> {
+  protected async getOneShallowOrFail(where: ObjectLiteral): Promise<T> {
     const found = await this.findOne({ where });
 
     if (!found) {
@@ -454,7 +454,7 @@ export class TypeOrmCrudService<T> extends CrudService<T> {
     return found;
   }
 
-  private prepareEntityBeforeSave(dto: DeepPartial<T>, parsed: CrudRequest['parsed']): T {
+  protected prepareEntityBeforeSave(dto: DeepPartial<T>, parsed: CrudRequest['parsed']): T {
     /* istanbul ignore if */
     if (!isObject(dto)) {
       return undefined;

